src/env.cc

```diff
@@ -547,6 +549,7 @@ void Environment::RunCleanup() {
     }
     CleanupHandles();
   }
+  GetAsyncRequest()->Uninstall();
```

Comment:
Can you move this to *before* a `CleanupHandles()` call, rather than at the end of this function? Otherwise the handle is still in closing state (rather than destroyed) when this function exits.